toolchain/check/testdata/basics/numeric_literals.carbon

```diff
@@ -95,12 +139,12 @@ let e: f64 = 5.0e39999999999999999993;
 // CHECK:STDOUT:   %int_8.b85: Core.IntLiteral = int_value 8 [concrete]
 // CHECK:STDOUT:   %int_9.988: Core.IntLiteral = int_value 9 [concrete]
 // CHECK:STDOUT:   %int_2147483647.d89: Core.IntLiteral = int_value 2147483647 [concrete]
-// CHECK:STDOUT:   %float.8ca: Core.FloatLiteral = float_value 0.90000000000000002 [concrete]
-// CHECK:STDOUT:   %float.fde: Core.FloatLiteral = float_value 8 [concrete]
-// CHECK:STDOUT:   %float.7a6: Core.FloatLiteral = float_value 80 [concrete]
-// CHECK:STDOUT:   %float.23b: Core.FloatLiteral = float_value 1.0E+7 [concrete]
-// CHECK:STDOUT:   %float.b98: Core.FloatLiteral = float_value 1.0E+8 [concrete]
-// CHECK:STDOUT:   %float.34d: Core.FloatLiteral = float_value 1.0E-8 [concrete]
+// CHECK:STDOUT:   %.7d0: Core.FloatLiteral = float_literal_value 9e-1 [concrete]
+// CHECK:STDOUT:   %.4bc: Core.FloatLiteral = float_literal_value 80e-1 [concrete]
+// CHECK:STDOUT:   %.24e: Core.FloatLiteral = float_literal_value 800e-1 [concrete]
```

Comment:
Strange result - the 80e-1 and 800e-1 rather than 8e1 and 8e2? I guess this falls out of something else, so probably not worth changing - just odd.